### PR TITLE
DOC: stats.jarque_bera: add semi-realistic example

### PR DESCRIPTION
**Pull request by mdhaber**
#### What does this implement/fix?
This adds a semi-realistic example to the documentation of `scipy.stats.jarque_bera`.

### CODE REVIEW

File: scipy/stats/_stats_py.py [hunk 1]
```diff
@@ -1817,20 +1817,96 @@ def jarque_bera(x, *, axis=None):
...
+    ...     s = stats.skew(x, axis=axis)
+    ...     k = stats.kurtosis(x, axis=axis)
+    ...     return x.shape[axis]/6 * (s**2 + k**2/4)
+    >>> res = stats.monte_carlo_test(x, stats.norm.rvs, statistic)
+    >>> fig, ax = plt.subplots()
+    >>> ax.hist(res.null_distribution, bins=np.linspace(0, 10, 50),
+    ...         density=True)
+    >>> ax.plot(jb_val, dist.pdf(jb_val))
+    >>> ax.legend(['Monte Carlo approximation (11 observations)',
+                   'aymptotic approximation (many observations'])
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:1899]:
```suggestion
    >>> ax.legend(['Monte Carlo approximation (11 observations)',
    ...            'aymptotic approximation (many observations'])
```

File: scipy/stats/_stats_py.py [hunk 2]
```diff
@@ -1817,20 +1817,96 @@ def jarque_bera(x, *, axis=None):
...
+    >>> dist.sf(res.statistic)
+    0.03045746622458189
+    >>> res.pvalue
+    0.03045746622458189
+
+    If the p-value is small - that is, if there is a low probability of
+    sampling data from a normally distributed population that produces such an
+    extreme value of the statistic - this may be taken as evidence against
+    the null hypothesis in favor of the alternative: the sample was not
+    drawn from a normal distribution.
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:1881]:
```suggestion
    the null hypothesis in favor of the alternative: the sample was not
    drawn from a normal distribution. The inverse is not true; that is,
    the test is not used to provide evidence for the null hypothesis.
```

File: scipy/stats/_stats_py.py [hunk 3]
```diff
@@ -1817,20 +1817,97 @@ def jarque_bera(x, *, axis=None):
...
+    The comparison is quantified by the p-value: the proportion of values in
+    the null distribution greater than or equal to the observed value of the
+    statistic.
+
+    >>> dist.sf(res.statistic)
+    0.03045746622458189
+    >>> res.pvalue
+    0.03045746622458189
+
+    If the p-value is small - that is, if there is a low probability of
```
Comment (tupui) [inline scipy/stats/_stats_py.py:1877]: Small being relative, I think we need to say with respect to which level (5%?)

File: scipy/stats/_stats_py.py [hunk 4]
```diff
@@ -1817,20 +1817,97 @@ def jarque_bera(x, *, axis=None):
...
+    >>> res = stats.monte_carlo_test(x, stats.norm.rvs, statistic)
+    >>> fig, ax = plt.subplots()
+    >>> ax.hist(res.null_distribution, bins=np.linspace(0, 10, 50),
+    ...         density=True)
+    >>> ax.plot(jb_val, dist.pdf(jb_val))
+    >>> ax.legend(['Monte Carlo approximation (11 observations)',
+    ...            'aymptotic approximation (many observations'])
+    >>> ax.set_title("Jarque-Bera Null Distribution")
+    >>> ax.set_xlabel("statistic")
+    >>> ax.set_ylabel("frequency density")
```
Comment (tupui) [inline scipy/stats/_stats_py.py:1903]: It might be confusing to have another y-label compared to the first figure.

File: scipy/stats/_stats_py.py [hunk 5]
```diff
@@ -1817,20 +1817,97 @@ def jarque_bera(x, *, axis=None):
...
+    observations. For small samples like ours, `scipy.stats.monte_carlo_test`
+    may provide a more accurate, albeit stochastic, approximation of the
+    exact p-value.
+
+    >>> def statistic(x, axis):
+    ...     s = stats.skew(x, axis=axis)
+    ...     k = stats.kurtosis(x, axis=axis)
+    ...     return x.shape[axis]/6 * (s**2 + k**2/4)
+    >>> res = stats.monte_carlo_test(x, stats.norm.rvs, statistic)
+    >>> fig, ax = plt.subplots()
```
Comment (tupui) [inline scipy/stats/_stats_py.py:1895]: We need to adjust a big so the legend does not overlap
```suggestion
    >>> fig, ax = plt.subplots(figsize=(8, 5))
```

**Review by tupui — COMMENTED** [review]
Nice and complete example 👍 I have a few suggestions.

File: scipy/stats/_stats_py.py [hunk 6]
```diff
@@ -1817,20 +1817,97 @@ def jarque_bera(x, *, axis=None):
...
+    0.03045746622458189
+    >>> res.pvalue
+    0.03045746622458189
+
+    If the p-value is small - that is, if there is a low probability of
+    sampling data from a normally distributed population that produces such an
+    extreme value of the statistic - this may be taken as evidence against
+    the null hypothesis in favor of the alternative: the sample was not
+    drawn from a normal distribution. The inverse is not true; that is,
+    the test is not used to provide evidence for the null hypothesis.
```
Comment (tupui) [inline scipy/stats/_stats_py.py:1882]: Maybe rephrase slightly to talk again about the weight.

File: scipy/stats/_stats_py.py (hunk 3 above)
Comment (mdhaber) [reply scipy/stats/_stats_py.py:1877]: That is always up to the researcher. I always avoid suggesting a level. But I will make it explicit that it is the researcher's choice and perhaps note the convention of a few fields.

File: scipy/stats/_stats_py.py [hunk 7]
```diff
@@ -1817,20 +1817,97 @@ def jarque_bera(x, *, axis=None):
...
+    the chi-squared distribution with two degrees of freedom.
+
+    >>> import matplotlib.pyplot as plt
+    >>> fig, ax = plt.subplots()
+    >>> dist = stats.chi2(df=2)
+    >>> jb_val = np.linspace(0, 10, 100)
+    >>> ax.plot(jb_val, dist.pdf(jb_val))
+    >>> ax.set_title("PDF of Jarque-Bera Null Distribution")
+    >>> ax.set_xlabel("statistic")
+    >>> ax.set_ylabel("probability density")
```
[the remark this answers is not in the document]
Comment (mdhaber) [reply scipy/stats/_stats_py.py:1865]: But a p-value is a probability, not a probably density. I could plot the survival function instead of the PDF.

File: scipy/stats/_stats_py.py (hunk 4 above)
Comment (mdhaber) [reply scipy/stats/_stats_py.py:1903]: If I change these to the survival function, I would not be uncomfortable regarding labeling both "p-value".

File: scipy/stats/_stats_py.py (hunk 3 above)
Comment (tupui) [reply scipy/stats/_stats_py.py:1877]: Totally, and in particule physics 3% might still be high. That's what I had in mind with saying that we need a reference to say "small"

File: scipy/stats/_stats_py.py [hunk 8]
```diff
@@ -1817,20 +1817,115 @@ def jarque_bera(x, *, axis=None):
...
+    ...     s = stats.skew(x, axis=axis)
+    ...     k = stats.kurtosis(x, axis=axis)
+    ...     return x.shape[axis]/6 * (s**2 + k**2/4)
+    >>> res = stats.monte_carlo_test(x, stats.norm.rvs, statistic)
+    >>> fig, ax = plt.subplots(figsize=(8, 5))
+    >>> ax.plot(jb_val, pdf)
+    >>> ax.hist(res.null_distribution, np.linspace(0, 10, 50),
+    ...         density=True)
+    >>> ax.legend(['aymptotic approximation (many observations)',
+                   'Monte Carlo approximation (11 observations)'])
```
Comment (tupui) [inline scipy/stats/_stats_py.py:1918]:
```suggestion
    >>> ax.legend(['aymptotic approximation (many observations)',
    ...            'Monte Carlo approximation (11 observations)'])
```

File: scipy/stats/_stats_py.py [hunk 9]
```diff
@@ -1817,20 +1817,115 @@ def jarque_bera(x, *, axis=None):
...
+    The comparison is quantified by the p-value: the proportion of values in
+    the null distribution greater than or equal to the observed value of the
+    statistic.
+
+    >>> fig, ax = plt.subplots(figsize=(8, 5))
+    >>> jb_plot(ax)
+    >>> pvalue = dist.sf(res.statistic)
+    >>> annotation = (f'p-value={pvalue:.6f}\n(shaded area)')
+    >>> arrowprops=dict(facecolor='black', width=1, headwidth=5, headlength=8)
+    >>> ax.annotate(annotation, (7.5, 0.01), (8, 0.05), arrowprops=arrowprops)
```
Comment (tupui) [inline scipy/stats/_stats_py.py:1879]: Let's keep the missing spaces in `arrowprops=dict` to not go overline.
And this should fix the ref guide check.
```suggestion
    >>> annotation = (f'p-value={pvalue:.6f}\n(shaded area)')
    >>> arrowprops=dict(facecolor='black', width=1, headwidth=5, headlength=8)
    >>> _ = ax.annotate(
    ...     annotation, (7.5, 0.01), (8, 0.05), arrowprops=arrowprops
    ... )
```

**Review by tupui — COMMENTED** [review]
Not sure why some figures are not showing up. Maybe because some code was wrong it did not like the whole example.

**Review by tupui — APPROVED** [review]
CI is coming back green and the figures looks great. I like the filling to get the p-value. Really nice 👍 I will merge once linter and ref guide come back green.
